### ISSUE TITLE
Raise an Exception of a task, actor, or placement groups is permanently infeasible

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#18835]
After https://github.com/ray-project/ray/pull/18724, we should be able to raise exceptions when a task, actor, or placement group becomes permanently infeasible. To do this, the autoscaler can periodically publish a list of "permanently infeasible" resource demands to the GCS via an RPC, and this list can be distributed across the cluster in resource poll requests from the GCS.

The errors raised would be as follows:
```
@ray.remote(num_gpus=999)
class A:
   def f(self):
      pass

@ray.remote(num_gpus=999)
def f():
   pass

ray.get(f.remote())
# -> raises UnschedulableError("No available node types can fulfill request {"GPU": 999}.")

a = A.remote()
ray.get(a.f.remote())
# -> raises UnschedulableError("No available node types can fulfill request {"GPU": 999}.")

pg = ray.placement_group([{"GPU": 999}])
ray.get(pg.ready())
# -> raises UnschedulableError("No available node types can fulfill request {"GPU": 999}.")

ray.get(f.options(placement_group=pg).remote())
# -> raises UnschedulableError("No available node types can fulfill request {"GPU": 999}.")
```

PRD doc: https://docs.google.com/document/d/1OT6m4xQDN8UtsBgnAMpX6nhXpNAfdeHJVve-iGhw1WI/edit

cc @edoakes @scv119 @richardliaw

**Comment by rkooo567** [comment]
```
pg = ray.placement_group([{"GPU": 999}])
ray.get(pg.ready())
# -> raises UnschedulableError("The cluster configuration cannot fulfill [{"GPU": 999}].")
```
@ericl do you also plan to mark "placement groups that are already deleted" as infeasible?

**Comment by ericl** [comment]
Yes we should do that too.

**Comment by rkooo567** [comment]
Btw, when do we plan to do this task?

**Comment by ericl** [comment]
@rkooo567 you're currently working on a design for this right? cc @richardliaw

**Comment by rkooo567** [comment]
Yes. I've been putting it off a little bit (focus on other tasks first) because of the feature freeze.

**Comment by rkooo567** [comment]
The initial draft; https://docs.google.com/document/d/158cRgisVt6JZ55ARckrRHDLjLJ6YA_9lnzN8bp9F6vU/edit

**Comment by xwjiang2010** [comment]
Hi @rkooo567 wondering if there is an update on this ticket? There is some feedback about insufficient resource messages can be confusing if autoscaler is enabled.

**Comment by ericl** [comment]
I think this is blocked on pending work from @wuisawesome refactoring the autoscaler interfaces.
